src/categorical_algebra/FinSets.jl

```diff
@@ -115,48 +121,48 @@ Base.collect(f::FinFunction) = force(f).func
   end
 end
 
-compose_impl(f::FinFunction, g::FinFunction) = g ∘ f
+compose_impl(f::FinFunction{S,T}, g::FinFunction{S,T}) where {S,T} = g ∘ f
 compose_impl(f::FinFunctionVector, g::FinFunctionVector) = g.func[f.func]
 
 # Limits
 ########
 
-function product(Xs::StaticVector{0,FinSet{Int}})
+function product(Xs::StaticVector{0,FinSet{Int,Int}})
```

Comment:
Since the second type parameter is basically an implementation detail, I suggest changing all these signatures to have form:

```julia
product(Xs::StaticVector{0,<:FinSet{Int}})
```